eos_data_distribution/ndn/http.py

```diff
@@ -57,14 +57,13 @@ def read_bytes_async():
     read_bytes_async()
 
 
-def fetch_http_headers(session, url):
+def fetch_http_headers(session, url, callback):
```

Comment:
Why are you trying to make header loading asynchronous? Why is this a hack? How does it differ from a full solution, and why is a hack justified at the moment? (I’m not arguing that a hack *isn’t* justified; I’d just like to see some explanation in the commit message.)

There’s not very much for me to go on with the review here.

One thought I have when looking at this: why does `Getter` exist? It’s a wrapper around libsoup which doesn’t add any additional functionality other than handling of the `Range` and `Content-Length` headers, which would be easy to do with helper functions. It’s only used from `http.Producer` — why not use libsoup directly there, and eliminate one level of Python callbacks on the hot path?